src/rpc/rawtransaction.cpp

```diff
@@ -63,28 +63,48 @@ class RPCRawTransaction_ECC_Init {
 };
 static RPCRawTransaction_ECC_Init ecc_init_on_load;
 
-void ScriptPubKeyToJSON(const CScript& scriptPubKey, UniValue& out, bool fIncludeHex)
+void ScriptPubKeyToJSONWithPrefix(const CScript& scriptPubKey, UniValue& out, bool fIncludeHex, const std::string prefix, bool is_parent_chain)
 {
     txnouttype type;
     vector<CTxDestination> addresses;
     int nRequired;
 
-    out.push_back(Pair("asm", ScriptToAsmStr(scriptPubKey)));
+    out.push_back(Pair(prefix + "asm", ScriptToAsmStr(scriptPubKey)));
     if (fIncludeHex)
-        out.push_back(Pair("hex", HexStr(scriptPubKey.begin(), scriptPubKey.end())));
+        out.push_back(Pair(prefix + "hex", HexStr(scriptPubKey.begin(), scriptPubKey.end())));
 
     if (!ExtractDestinations(scriptPubKey, type, addresses, nRequired)) {
-        out.push_back(Pair("type", GetTxnOutputType(type)));
+        out.push_back(Pair(prefix + "type", GetTxnOutputType(type)));
```

Comment:
there could be an odd case where this line is hit, yet it still claims return type of P2PK, where the pubkey is invalid... that's kind of odd. I'll check upstream.